### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Progress2002**
### Hello Reviewer,

This is @Progress2002  and @shafiuyushawu, and this milestone fulfills below requirements:

 - [ ] Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.

    - Do not use the bullet gem in this exercise - instead, analyze the logs as shown in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
    - **In your PR description include the screenshots of your app's logs before and after the fix**
    _Before_
user3-before

        _After_
   usr3-after

- [ ] Integration specs

  - Use Capybara to write integration tests for each view in your project.
  - Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.
  
  - User index page:
    - I can see the username of all other users.
    - I can see the profile picture for each user.
    - I can see the number of posts each user has written.
    - When I click on a user, I am redirected to that user's show page.
  
  - user show page:
    - I can see the user's profile picture.
    - I can see the user's username.
    - I can see the number of posts the user has written.
    - I can see the user's bio.
    - I can see the user's first 3 posts.
    - I can see a button that lets me view all of a user's posts.
    - When I click a user's post, it redirects me to that post's show page.
    - When I click to see all posts, it redirects me to the user's post's index page.
  
  - User post index page:
    - I can see the user's profile picture.
    - I can see the user's username.
    - I can see the number of posts the user has written.
    - I can see a post's title.
    - I can see some of the post's body.
    - I can see the first comments on a post.
    - I can see how many comments a post has.
    - I can see how many likes a post has.
    - I can see a section for pagination if there are more posts than fit on the view.
    - When I click on a post, it redirects me to that post's show page.
  
  - Post show page:
    - I can see the post's title.
    - I can see who wrote the post.
    - I can see how many comments it has.
    - I can see how many likes it has.
    - I can see the post body.
    - I can see the username of each commentor.
    - I can see the comment each commentor left.

### CODE REVIEW

File: spec/views_integration_test/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      page.has_content?(comment.text)
+    end
+    it 'I can see how many comments a post has.' do
+      visit user_posts_path(user.id)
+      page.has_content?(subject.comment_counter)
+    end
+    it 'When I click on a post, it redirects me to that posts show page.' do
+      visit user_posts_path(user.id)
+      click_on 'First post'
+      visit user_post_path(user.id, subject.id)
```
Comment (DuaneDave) [inline spec/views_integration_test/post_spec.rb:42]: Good job writing integration for this page, however, you have missed a couple of other tests such as:

- Tests that assert you can see how many likes a post has.
- Tests that assert you can see a section for pagination if there are more posts than fit on the view.

In order to satisfy the requirements for this milestone, consider writing the above-mentioned tests

**Review by DuaneDave — CHANGES_REQUESTED** [review]
Hi @shafiuyushawu  and @Progress2002  👋 , 

Good job so far on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Correct use of git-flow
✔️ You did a good job writing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @Progress2002, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views_integration_test/post_spec.rb (hunk 1 above)
Comment (Progress2002) [reply spec/views_integration_test/post_spec.rb:42]: Hello, @DuaneDave, thanks for reviewing my project. The above-mentioned test cases you suggested for me to implement already exist in the file.

**Comment by jerryowusu** [comment]
Hello @Progress2002, Your project have already been approved.

**Comment by Progress2002** [comment]
Yes but it's still showing changes required in my dashboard

**Comment by nicupop729** [comment]
@shafiuyushawu please note that this is a pair-programming activity where the students need to ask for review individually with their own PR.

The approval given by @DeliceLydia will be reverted in the dashboard for @shafiuyushawu.

In order to proceed further to the next milestone, @shafiuyushawu, kindly ask for another review using a PR from your own repository. Please don't forget to collaborate in this activity with your pair-programming partner.

We apologize for any inconvenience.

Good luck!

**Pull request merged**